### ISSUE TITLE
Incorrect String Formating for type Lx (LaTeX,SIunitx)

### ISSUE THREAD

**Issue by PorcelainMouse** [hgrecco/pint#814]
I think I have identified a bug with the way that both `pint` and `uncertainties` implement the Lx LaTeX with SIunitx package.

Symptom 1: uncertainty notation () (parentheses) used incorrectly.

> print('{:Lx}'.format( (10.5 * ureg.m).plus_minus(1) ))
>\SI[separate-uncertainty=true]{10.5(1.0)}{\meter}

This is a super nice feature of `pint`, but () notation takes a string of digits that represent the uncertainty of the *least significant digits* of the value and does not accept a decimal.  I would think an easy fix would be to use ' +/- ' notation alternative of \SI to join value and uncertainty, which would accept uncertainty values like this.  LaTeX throws:

> ! siunitx error: "invalid-token-in-uncertainty"
> ! Invalid uncertainty in numerical input '10.5(1.0)'.

Symptom 2: uncertainty notation ' +/- ' used incorrectly

> print('{:Lx}'.format(unc.ufloat(10.5,1) * ureg.m))
> \SI[]{10.5+/-1.0}{\meter} 

I don't understand why `uncertainties` module implements Lx differently, but it is also producing incorrect code.  Should I report this separately to Eric?  `uncertainties` uses the +/- notation, but without the required white space surrounding it.  LaTeX throws:

> ! siunitx error: "misplaced-sign-token"
> ! Misplaced sign token '10.5+'.

I'd be happy to try to fix and send patches, but I want to know if that would be helpful and, if so, if you have an idea about how exactly you would like the bug fix to be implemented.

**Comment by hgrecco** [comment]
I would be nice to have bug fix. My only requirement is that you try to keep most changes (all?) within formatting.py so printing is isolated.

**Comment by PorcelainMouse** [comment]
I took a gander; getting the lay of the land.

**Comment by PorcelainMouse** [comment]
Hmm, I've been looking at this for a while now.  I think I have convinced myself that pint is using uncertainties to do this formatting expansion.  So, after all that, I think I need to refer this to uncertainties, after all.

I see that you have added a unit test for the formats, though, which is awesome.  I started writing one back in July, but you are now  way past what I was writing.

**Comment by PorcelainMouse** [comment]
Okay, so __format__ has this comment:

`the uncertainties module supports formatting
...
which siunitx actually accepts as input`

This is incorrect.  siunitx is more restrictive than what uncertainties outputs in it's "Shorthand" format (type code 'S').  So, I'm confident this is the root cause of the problem.

I'm still not sure how best to fix it.  I could send you a PR with some new code to do it in Pint.  Or, I could try to patch uncertainties to support 'Lx', officially, and then Pint could be pass through, which is in keeping with the current implementation.  Let me know if you have any thoughts.  Perhaps I will do both.  Both should support 'Lx', if possible, I think.  Until uncertainties supports it, Pint will have to do it itself, so, in the meantime, I might as well.

**Comment by PorcelainMouse** [comment]
Hey, so after two days of work, I managed to create a special case for 'Lx' in measurement format, only to realize that I had broken a bunch of your tests.  I'm really lost in the code; I just don't understand enough of the existing design to have any confidence in my approach.  I'm want to help, but I feel like there is a lot of internal design that is not apparent, even after reading a lot of the code.

If you have any advice, I would appreciate it.

**Comment by PorcelainMouse** [comment]
After all that, I had a look at uncertainties source code and noticed something useful.  I might be on to something...

**Comment by PorcelainMouse** [comment]
Well, I think I have something work showing you, at least.  Please have a look at my PR.

**Comment by PorcelainMouse** [comment]
Slightly off topic: what do you think about a new string format conversion type to produce uncertain values in a "natural" precision, one that is based on the uncertainty, not arbitrary precision.  I mentioned this to the 'uncertainties' package maintainer, too, but it fits just as well in 'pint', I think.  Just floating the idea; I might be able to code it, if it sounds interesting to you.  This bug go me thinking: there are so many conversion types because it's so hard to know how the user might want to present the data; but, uncertainty actually determines the significant portion of the nominal value, and that is a "natural" way to define a default string representation.  Let me know if that seems worth pursuing.

**Comment by PorcelainMouse** [comment]
Okay, so, apparently I was wrong.  The uncertainties package's implementation of the compact uncertainty format ('S' type conversion) is correct. See the bug I referenced, which is now closed.

However, this only makes the patch I submitted more important because now we know 'uncertainties' is never going to be changed to match SIunitx current implementation.  So, until SIunitx is fixed, 'pint' will be solely responsible for 'Lx' conversion type.
